openpgp/keys.go

```diff
@@ -106,53 +112,64 @@ func (e *Entity) EncryptionKey(now time.Time) (Key, bool) {
 
 	if candidateSubkey != -1 {
 		subkey := e.Subkeys[candidateSubkey]
-		return Key{e, subkey.PublicKey, subkey.PrivateKey, subkey.Sig}, true
+		encryptionKey, found = Key{e, subkey.PublicKey, subkey.PrivateKey, subkey.Sig}, true
+		return
```

Comment:
To me, the old pattern of returning directly, rather than assigning and then returning, is a little bit more clear. Could you revert these changes? (Also in `SigningKey`)